plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt

```diff
@@ -107,3 +113,94 @@ class SourcesetDependencyAppender(val context: DokkaContext) : PageTransformer {
         )
     }
 }
+
+class SearchbarDataInstaller(val context: DokkaContext): PageTransformer {
+    private val pageList = mutableMapOf<String, Pair<String, String>>()
+    lateinit var locationProvider : LocationProvider
+
+        override fun invoke(input: RootPageNode): RootPageNode {
+            locationProvider = context.plugin<DokkaBase>().querySingle { locationProviderFactory }
+                .getLocationProvider(input)
+            traverse(input)
+            val data = RendererSpecificResourcePage(
+                name = "scripts/pages.js",
+                children = emptyList(),
+                strategy = RenderingStrategy.Write(generatePagesList())
+            )
+
+            return input.modified(
+                children = input.children + data
+            )
+    }
+
+    private fun traverse(page:PageNode): Unit {
+        if(page is ContentPage && page !is ModulePageNode && page !is PackagePageNode) processPage(page)
+        page.children.forEach(::traverse)
```

Comment:
[Minor] Imho a tailrec function with accumulator would look better than side effect

Reply:
I deleted this recursion as it was failing with strange exceptions when tried to run it after rendering